predicate/src/lib.rs

```diff
@@ -450,12 +451,45 @@ impl PredicateBuilder {
     }
 }
 
-// A representation of the `BinaryExpr` variant of a Datafusion expression.
+// Wrapper around `Expr::BinaryExpr` where left input is known to be
+// single Column reference to the `_value` column
 #[derive(Clone, Debug, PartialEq, PartialOrd)]
-pub struct BinaryExpr {
-    pub left: Column,
-    pub op: Operator,
-    pub right: Expr,
+pub struct ValueExpr {
+    expr: Expr,
+}
+
+impl ValueExpr {
+    /// tries to create a new ValueExpr. If `expr` follows the
+    /// expected pattrn, returns Ok(Self). If not, returns Err(expr)
+    pub fn try_new(expr: Expr) -> std::result::Result<Self, Expr> {
+        if let Expr::BinaryExpr {
+            left,
+            op: _,
+            right: _,
+        } = &expr
+        {
+            if let Expr::Column(inner) = left.as_ref() {
+                if inner.name == VALUE_COLUMN_NAME {
+                    return Ok(Self { expr });
+                }
+            }
+        }
+        Err(expr)
+    }
+
+    /// Returns a new Expr with th reference to `_value` replaced with the specified value and
```

Comment:
This comment looks incomplete and has a typo :)

```suggestion
    /// Returns a new Expr with the reference to `_value` replaced with the specified value and...???
```